pkg/registry/registry.go

```diff
@@ -7,11 +7,27 @@ import (
 
 //Interface to maintain cluster configuration for a sharding manager identity
 type ClusterConfiguration interface {
-	GetClustersByShardingManagerIdentity(ctx context.Context, shardingManagerIdentityName string) ([]model.Cluster, error)
-	BulkSyncByShardingManagerIdentity(ctx context.Context, shardingManagerIdentityName string) ([]model.Cluster, error)
+	GetClustersByShardingManagerIdentity(ctx context.Context, shardingManagerIdentityName string) ([]*model.ClusterConfig, error)
+	BulkSyncByShardingManagerIdentity(ctx context.Context, shardingManagerIdentityName string) ([]*model.ClusterConfig, error)
 }
 
 //Interface to maintain identity configuration for a cluster
 type IdentityConfiguration interface {
-	GetIdentitiesByCluster(ctx context.Context, clusterName string) ([]model.Identity, error)
+	GetIdentitiesByCluster(ctx context.Context, clusterName string) ([]*model.IdentityConfig, error)
+}
+
+type RegistryClient struct {
```

Comment:
Can this be private?

Reply:
Used outside of the package so needs to be public